packages/firestore/src/util/async_queue.ts

```diff
@@ -38,20 +50,42 @@ class DelayedOperation<T> implements CancelablePromise<T> {
   private readonly deferred = new Deferred<T>();
 
   private constructor(
-    private asyncQueue: AsyncQueue,
-    private op: () => Promise<T>
+    private readonly asyncQueue: AsyncQueue,
+    readonly timerId: TimerId,
+    readonly targetTime: number,
```

Comment:
TargetTime is pretty unspecific. Should me make this a Date or call this `targetTimeMs`?